modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/MenuShelfItems.kt

```diff
@@ -0,0 +1,143 @@
+package au.com.shiftyjelly.pocketcasts.player.view.shelf
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import au.com.shiftyjelly.pocketcasts.compose.AppTheme
+import au.com.shiftyjelly.pocketcasts.compose.preview.ThemePreviewParameterProvider
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastEpisode
+import au.com.shiftyjelly.pocketcasts.player.viewmodel.ShelfViewModel
+import au.com.shiftyjelly.pocketcasts.player.viewmodel.ShelfViewModel.Companion.moreActionsTitle
+import au.com.shiftyjelly.pocketcasts.player.viewmodel.ShelfViewModel.Companion.shortcutTitle
+import au.com.shiftyjelly.pocketcasts.player.viewmodel.ShelfViewModel.UiState
+import au.com.shiftyjelly.pocketcasts.preferences.model.ShelfItem
+import au.com.shiftyjelly.pocketcasts.preferences.model.ShelfTitle
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import java.util.Date
+import sh.calvin.reorderable.ReorderableItem
+import sh.calvin.reorderable.rememberReorderableLazyListState
+
+@Composable
+fun MenuShelfItems(
+    shelfViewModel: ShelfViewModel,
+    normalBackgroundColor: Color = Color.Transparent,
+    selectedBackgroundColor: Color = Color.Black,
+    onClick: ((ShelfItem, Boolean) -> Unit)? = null,
+) {
+    val uiState by shelfViewModel.uiState.collectAsStateWithLifecycle(null)
+    uiState?.let { state ->
+        if (state.episode == null) return
```

Comment:
Why is this check needed? I see that `ShelfItemRow` accepts `BaseEpisode?` so it shouldn't be an issue to pass in `null`.

Reply:
Removed in fcc3ae4